tests/main/auto-refresh-retry/task.yaml

```diff
@@ -0,0 +1,75 @@
+summary: Check that auto-refresh is retried after a permanent network error
+
+# FIXME: core18 right now does not have a canonical model. This means
+# that it will not get a serial. We code in devicestate.go that will only
+# try to auto-refresh after the systems has tried at least 3 times to get
+# a serial. But this means this test will timeout because the first retry
+# for the serial happens after 5min, then 10min, then 20min.
+systems: [-ubuntu-core-18-*]
+
+prepare: |
+    snap install --devmode jq
+    cp /lib/systemd/system/snapd.service /lib/systemd/system/snapd.service.bak
+
+restore: |
+    mv -f /lib/systemd/system/snapd.service.bak /lib/systemd/system/snapd.service
+
+debug: |
+    ip netns list
+    ip netns pids testns || true
+
+execute: |
+    # shellcheck source=tests/lib/journalctl.sh
+    . "$TESTSLIB/journalctl.sh"
+
+    echo "Install a snap from stable"
+    snap install test-snapd-tools
+  
+    snap set core refresh.schedule="0:00-23:59"
+    systemctl stop snapd.{service,socket}
+
+    echo "Modify the snap to track the edge channel"
+    jq ".data.snaps[\"test-snapd-tools\"].channel = \"edge\"" /var/lib/snapd/state.json > /var/lib/snapd/state.json.new
```

Comment:
I see we've stopped snapd now but I wonder, what would have happened if snapd was still running and we were to `snap switch test-snapd-tools --channel=edge`